tests/collections/asr/test_asr_datasets.py

```diff
@@ -125,3 +158,164 @@ def test_tarred_bpe_dataset(self, test_data_dir):
+
+        num_samples = 10
+        batch_size = 2
+        device = 'gpu' if torch.cuda.is_available() else 'cpu'
+
+        tokenizer_path = os.path.join(test_data_dir, "asr", "tokenizers", "an4_wpe_128", 'vocab.txt')
+        tokenizer = tokenizers.AutoTokenizer(pretrained_model_name='bert-base-cased', vocab_file=tokenizer_path)
+
+        with tempfile.NamedTemporaryFile(mode='w', encoding='utf-8') as f:
+            with open(manifest_path, 'r') as m:
+                for ix, line in enumerate(m):
+                    if ix >= num_samples:
+                        break
+
+                    line = line.replace("tests/data/", "tests/.data/").replace("\n", "")
+                    f.write(f"{line}\n")
+                    num_samples += 1
+
+            f.seek(0)
+
+            dataset = AudioToBPEDALIDataset(
+                manifest_filepath=f.name,
+                tokenizer=tokenizer,
+                device=device,
+                batch_size=batch_size,
+                max_duration=16.0,
+                shuffle=False,
+            )
+
+            assert len(dataset) == 5  # num batches
+            count = 0
+            original_transcripts = []
+            for batch in dataset:
+                transcripts = batch[2]  # transcript index in DALIOutputs
+                transcripts_lengths = batch[3]  # transcript length index in DALIOutputs
+                transcripts = [
+                    decode_subwords(transcript, transcripts_length, tokenizer=tokenizer)
+                    for transcript, transcripts_length in zip(transcripts, transcripts_lengths)
+                ]
+                original_transcripts.extend(transcripts)
+                count += len(transcripts)
+            assert count == 10
+
+            # Repeat, now with shuffle enabled
+            f.seek(0)
+
+            dataset = AudioToBPEDALIDataset(
+                manifest_filepath=f.name,
+                tokenizer=tokenizer,
+                device=device,
+                batch_size=batch_size,
+                max_duration=16.0,
+                shuffle=True,
+            )
+
+            assert len(dataset) == 5  # num batches
+            count = 0
+            shuffled_transcripts = []
+            for batch in dataset:
+                transcripts = batch[2]  # transcript index in DALIOutputs
+                transcripts_lengths = batch[3]  # transcript length index in DALIOutputs
+                transcripts = [
+                    decode_subwords(transcript, transcripts_length, tokenizer=tokenizer)
+                    for transcript, transcripts_length in zip(transcripts, transcripts_lengths)
+                ]
+                shuffled_transcripts.extend(transcripts)
+                count += len(transcripts)
+            assert count == 10
+
+            sample_matches = 0
+            for orig, shuffled in zip(original_transcripts, shuffled_transcripts):
```

Comment:
Similar remarks as in the previous test regarding the results validation.

Reply:
Adressed.